subsys/bluetooth/audio/tbs.c

```diff
@@ -308,16 +308,21 @@ static uint8_t next_free_call_index(void)
 {
 	for (int i = 0; i < CONFIG_BT_TBS_MAX_CALLS; i++) {
 		static uint8_t next_call_index = 1;
-		const struct bt_tbs_call *call = lookup_call(next_call_index);
+		const struct bt_tbs_call *call;
 
-		if (call == NULL) {
-			return next_call_index++;
+		if (next_call_index == 0) {
+			/* call_index = 0 reserved for outgoing calls */
+			next_call_index = 1;
 		}
 
+		call = lookup_call(next_call_index);
+
+		/* For each new call, the call index should be incremented */
 		next_call_index++;
-		if (next_call_index == UINT8_MAX) {
-			/* call_index = 0 reserved for outgoing calls */
-			next_call_index = 1;
+
+		if (call == NULL) {
+			/* call == NULL is true for (next_call_index - 1)*/
+			return (next_call_index - 1);
```

Comment:
Wouldn't it make more sense to increment `next_call_index` after the check? That way you can return `next_call_index` instead of `next_call_index - 1`